### DISCUSSION TITLE
How to work with uploaded image/ file

### DISCUSSION THREAD

**Discussion by kellertobias** [keystonejs/keystone#7308] [Questions] [answered]
I am writing a simple backend for a blog using keystone.js. I need to be able to upload thumbnails for images; While the image package solves the upload part, I then need to work with the uploaded image to create multiple versions for it.

Let's assume, I have the following async function `generateThumbnails(src: string): Promise<{thumb: string, poster: string}>` and I want to call this function once the image was uploaded and then store the thumb and poster URLs in two different fields in the schema, how can I "hook" the image uploaded event on the server?

**Comment by kellertobias** [comment]
It also might make sense to be able to change the URL that is stored for a picture, e.g. because we want to upload it to S3 after converting

**Comment by emmatown** [comment, marked as answer]
You could add a list level hook like this: 

```ts
hooks: {
  async resolveInput({ resolvedData }) {
    if (resolvedData.avatar.id !== undefined) {
      const { thumb, poster } = createThumnails(
        `${storagePath}/${resolvedData.avatar.id}.${resolvedData.avatar.extension}`
      );
      return {
        ...resolvedData,
        thumb,
        poster,
      };
    }

    return resolvedData;
  },
},
```

(this assumes you just want to  `thumb` and `poster` as strings in `text` fields)
